src/test/java/com/solbeg/MainTest.java

```diff
@@ -0,0 +1,231 @@
+            User.builder().id(3L).name("Nolan").email("nolandonovan@gmail.com").balance(BigDecimal.valueOf(13889)).build(),
+            User.builder().id(4L).name("Lucas").email("lucas.lynn@yahoo.com").balance(BigDecimal.valueOf(16980)).build());
+
+    private static final List<UserBankAccount> bankAccounts = Arrays.asList(
+            new UserBankAccount(new User(1L, "Justin", "justin.butler@gmail.com", BigDecimal.valueOf(172966)), BigDecimal.valueOf(172966)),
+            new UserBankAccount(new User(2L, "Olivia", "cardenas@mail.com", BigDecimal.valueOf(38029)), BigDecimal.valueOf(38029)),
+            new UserBankAccount(new User(3L, "Nolan", "nolandonovan@gmail.com", BigDecimal.valueOf(13889)), BigDecimal.valueOf(13889)),
+            new UserBankAccount(new User(4L, "Lucas", "lucas.lynn@yahoo.com", BigDecimal.valueOf(16980)), BigDecimal.valueOf(16980)));
+
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private final PrintStream originalOut = System.out;
+
+
+    @Test
+    void ifStringIsPresent_isPresentShouldReturnTrueIsEmptyFalse() {
+
+        assertTrue(Main.optionalOfString("text").isPresent());
+        assertFalse(Main.optionalOfString("text").isEmpty());
+    }
+
+    @Test
+    void ifStringIsNull_isPresentShouldReturnFalseIsEmptyTrue() {
+
+        assertFalse(Main.optionalOfString(null).isPresent());
+        assertTrue(Main.optionalOfString(null).isEmpty());
+    }
+
+    @Test
+    void ifBalanceZero_balanceShouldBeEqualsAmount() {
+
+        BigDecimal amount = new BigDecimal("100");
+        User user = Users.generateUser();
+        user.setBalance(new BigDecimal("0"));
+        UserProvider userProvider = () -> Optional.of(user);
+
+        Main.deposit(userProvider, amount);
+
+        assertEquals(amount, user.getBalance());
+    }
+
+    @Test
+    void ifIsNull_shouldTrowNPE() {
+
+        User user = Users.generateUser();
+
+        assertTrue(Main.optionalOfUser(user).isPresent());
+    }
+
+    @Test
+    void ifUserProviderDoNotProvideUser_shouldReturnDefaultUser() {
+
+        User defaultUser = Users.generateUser();
+        UserProvider userProvider = Optional::empty;
+
+        assertEquals(defaultUser, Main.getUser(userProvider, defaultUser));
+    }
+
+    @BeforeEach
+    public void setUpStreams() {
+
+        System.setOut(new PrintStream(outContent));
+    }
+
+    @AfterEach
+    public void restoreStreams() {
+
+        System.setOut(originalOut);
+    }
+
+    @Test
+    void ifUserIsNotProvided_shouldPrintMessage() {
+
+        String expectedMessage = "No user found\r\n";
+        UserProvider userProvider = Optional::empty;
+        UserService userService = new UserService() {
+            @Override
+            public void processUser(User user) {
+            }
+        };
+
+        Main.processUser(userProvider, userService);
+
+        assertEquals(expectedMessage, outContent.toString());
```

Comment:
no need to test outContent you should only verify if this method was called once
take a look at Mockito verify

Reply:
fixed